### PR TITLE
got rid of most os.path calls

### PR DESCRIPTION
**Pull request by scifi6546**
### What does this PR do?
This Pr cleans up most of the os.path calls and replaces them with calls from pathlib.Path, as requested in issue https://github.com/obspy/obspy/issues/2751

### Why was it initiated?  Any relevant Issues?

The pr was initiated to answer issue: https://github.com/obspy/obspy/issues/2751.

### CODE REVIEW

**Comment by scifi6546** [comment]
How should I run a lint on my own and should I add a separate pr to add running a linter to CONTRIBUTING.md?

**Comment by d-chambers** [comment]
Hi @scifi6546 ,

To run flake8 I recommend just installing and running [pre-commit](https://pre-commit.com/).
In your obspy directory run the following:

```bash
pip install pre-commit
pre-commit install
pre-commit run --all
```
After that pre-commit will run flake8 before it allows you to make any commits.

> should I add a separate pr to add running a linter to CONTRIBUTING.md

I wouldn't bother changing this file in this PR. If you want to cleanup some markdown files it would be better in a different PR.

**Comment by scifi6546** [comment]
Ok thanks. I think I got all of the linter warnings fixed now.

**Comment by scifi6546** [comment]
Ok the failure looks like it was caused by me not importing `os` for a doctest. Can you rerun the unit tests?

**Comment by d-chambers** [comment]
Sure. Its is annoying you can't trigger them by pushing. I should only have to give you permission once...

**Comment by scifi6546** [comment]
ok it looks like there is a broken figure. How should I find the image that is broken?

File: obspy/scripts/sds_html_report.py [hunk 1]
```diff
@@ -215,7 +214,7 @@ def _format_html(args, data_normal, data_outdated):
     html_legend['ok_info'] = "All checks pass"
     html_legend['outdated_info'] = (
         "No data within {check_back_days:d} days").format(**vars(args))
-    html_legend['output_basename'] = os.path.basename(args.output)
+    html_legend['output_basename'] = Path(args.output).parent
```
Comment (d-chambers) [inline obspy/scripts/sds_html_report.py:217]: This needs to be 
```python
html_legend['output_basename'] = Path(args.output).name
```
then it should fix the failing test.

**Comment by d-chambers** [comment]
This looks good to me, anything else you want to do with it? Did you get your name added to the contributors list?

**Comment by scifi6546** [comment]
Ahh yea adding my name to the contributors list is a good idea. Got that done and there is nothing else I want to do with the commit.

**Comment by jkmacc-LANL** [comment]
I learned a lot about pathlib by looking at this PR; thanks!  Would it also be good to update the changelog with these changes?

**Comment by d-chambers** [comment]
@jkmacc-LANL Good point, a changelog entry would be smart, even though the changes are entirely internal, in case there are any unintended consequences. After that @scifi6546 I will give the changes one more look over and merge it.

**Comment by scifi6546** [comment]
all right added my changes to the change log.

**Comment by d-chambers** [comment]
+TESTS:ALL

**Review by d-chambers — APPROVED** [review]
Looks good to me, thanks for taking time to cleanup the codebase a bit.

**Comment by scifi6546** [comment]
awesome!

**Pull request merged**